### ISSUE TITLE
SIGSEGV in Monero.com Wallet (Android 14, GrapheneOS, Pixel 8 Pro, MTE)

### ISSUE THREAD

**Issue by TempUser13** [cake-tech/cake_wallet#1199]
Simply opening the Monero.com wallet on Pixel 8 Pro using GrapheneOS Version: [2023112600](https://grapheneos.org/releases#2023112600) with MTE enabled makes a SIGSEGV

```
type: crash
osVersion: google/husky/husky:14/UD1A.231105.004/2023112600:user/release-keys
uid: 1110181 (u:r:untrusted_app_32:s0:c181,c256,c523,c768)
cmdline: com.monero.app
processUptime: 2s

signal: 11 (SIGSEGV), code 9 (SEGV_MTESERR), faultAddr 700d84eecaa261c
threadName: 1.ui
MTE: enabled

backtrace:
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk!libcw_monero.so (TransactionInfoRow::TransactionInfoRow(Monero::TransactionInfo*)+160, pc 83c100)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk!libcw_monero.so (transactions_get_all+128, pc 83bfd4)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61558c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 764e44)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 764b84)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 764aac)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 764a24)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 764728)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 803e50)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc e54c10)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 6230a8)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623204)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 62397c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623690)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 621bd0)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61d66c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc e55274)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61bf68)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc e54c10)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 6230a8)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623204)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 62397c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623690)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 621bd0)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61d66c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc e55274)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61bf68)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc e54c10)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 6230a8)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623204)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 62397c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623690)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 621bd0)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61d66c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc e55274)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61bf68)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc e54c10)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 6230a8)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623204)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 62397c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623690)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 621bd0)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61d66c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc e55274)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61bf68)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc e54c10)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 6230a8)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623204)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 62397c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 623690)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 621bd0)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61d66c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61d5d8)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 624038)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 624138)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc d76930)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc d7af58)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc d75638)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61cd3c)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61cc24)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 61cbe8)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk (pc 6173e8)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk!libflutter.so (pc 7076c0)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk!libflutter.so (pc 83a514)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk!libflutter.so (pc 5ccb44)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk!libflutter.so (pc 352c28)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk!libflutter.so (pc 355fcc)
    /system/lib64/libutils.so (android::Looper::pollInner(int)+1276, pc 19e8c)
    /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+124, pc 1992c)
    /system/lib64/libandroid.so (ALooper_pollOnce+100, pc 222b4)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk!libflutter.so (pc 3560d8)
    /data/app/~~0-MmZO_sU_sMkkc6NXRfPQ==/com.monero.app-lURVrWSIEGX14i8xoSBV5A==/base.apk!libflutter.so (pc 354710)
    /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204, pc db00c)
    /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64, pc 6fd30)```

**Comment by tuxpizza** [comment]
Thanks for the report. Does it just crash over and over or does it eventually open? And disabling MTE makes it work as expected?

**Comment by TempUser13** [comment]
Yes, it'll crash until MTE is disabled then it works as expected

**Comment by tuxpizza** [comment]
Hi, could you please try this on the latest version (1.8.0)?
I tested this myself on a P8P with MTE enabled using that version and I was not able to reproduce.

**Comment by TempUser13** [comment]
I still get this on 1.8.0, I installed in a new profile and everything worked with MTE until the wallet has a balance then it'll crash on startup. I also noticedthat the page before you are shown the seed, it'll say 25 even with polyseed enabled.

**Comment by TempUser13** [comment]
Is there any chance for the next release to bring per architecture built apks rather than a single universal apk too? #1047

**Comment by tuxpizza** [comment]
> Is there any chance for the next release to bring per architecture built apks rather than a single universal apk too? #1047

I'll talk to the dev team about this before the next release (4.13.0)

**Comment by tuxpizza** [comment]
Fixed in 4.19.0
